### DISCUSSION TITLE
How to do a PUT request from a button?

### DISCUSSION THREAD

**Discussion by 1MuhammadYahya** [sveltejs/kit#7821] [Q&A] [answered]
I am fairly confused on how to convert an endpoint PUT function to use actions such that it can be called from a button.
My usecase consisted of setting a cookie for preserving dark-theme so that the initial SSR renders the page in the preferred theme. Previously I would just use 
`
export const PUT: RequestHandler = async ( {request} ) => {
// Validate Request
    return {
        headers: {
            'Set-Cookie': 'theme=${theme}; SameSite=Strict; Secure; HttpOnly;'
        }
    }
}
`
Inside +page.server.ts

However ever since the intorduction of Actons, I have no idea on how to convert this piece of code to the newer format. Am i missing something? Any help would be appreciated.
Thanks!

**Comment by NormandoHall** [comment, marked as answer]
**Form Actions** are always **POST**. If you need a **PUT** endpoint, you can do it as https://kit.svelte.dev/docs/routing#server with `server.ts`
